### ISSUE TITLE
pollingInterval not cancelled if request is outstanding

### ISSUE THREAD

**Issue by arbesfeld** [heroku/react-refetch#166]
I have built a decorator like this:

```
export default ({ LoadingView = undefined, refreshInterval = 0 } = {}) =>
  WrappedComponent => ({ params }) => ({
    fetchURL: {
      url: `api/`,
      refreshInterval,
    },
  }(({
  fetchURL,
  ...rest
}) => {
    function onFulfillment(data) {
      return <WrappedComponent {...rest} data={data} />;
    }

    return (
    <PromiseStateContainer
      ps={fetchURL}
      onFulfillment={onFulfillment}
      onPending={LoadingView}
    />
    );
  }));
```

And using it like this:

```
import decorator from '../decorator';

@decorator({ pollingInterval: 2000 })
class MyClass extends Component {}
```

However, if I unmount the component while a poll is pending, then the polling continues indefinitely.

**Comment by arbesfeld** [comment]
@ryanbrainard - this appears to happen whenever the first url doesn't "resolve", and then the component is unmounted

**Comment by ryanbrainard** [comment]
@arbesfeld Thanks for the report. I haven't tried reproducing yet, but sounds plausible in the non-resolving case. Marking as a bug.

**Comment by TacticalCoding** [comment]
I have noticed this behavior as well. I'm using react-router, and if I navigate to a new page before an API request has returned from react-refetch, the polling is not cancelled from the previous page we were on and I see requests for both pages.

**Comment by ryanbrainard** [comment]
The fix from @TacticalCoding has been pre-released in [v1.0.1-0](https://github.com/heroku/react-refetch/releases/tag/v1.0.1-0). Please try it out! Will cut a patch release if all goes well.

**Comment by 3ll3d00d** [comment]
This fixes it in my app. When will you publish an official release?
